### PR TITLE
HUBS-1695 | Add example documentation for database specific VDB provi…

### PR DESCRIPTION
**Pull request by Uddipaan-Hazarika**
# Context:
Add example documentation for database specific VDB provision parameters.

# Problem: It was hard to figure out the parameters needed to provision a specific database with a huge list of parameters available to us.

# Solution: We provided some set of known databases with terraform files that can be used to provision their respective VDBs.

# Testing: These MD files can be tested visually in a terraform supported [doc preview tool](https://registry.terraform.io/tools/doc-preview)

### CODE REVIEW

File: docs/guides/provider_guide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Guide: <guide name> Delphix Provider Guide
```
Comment (nick-mathison) [inline docs/guides/provider_guide.md:1]: Do we have two headers for this section? Is that needed?

Does the first one appear in the Table of Contents on the left hand side? The markdown tester doesn't show it.

File: docs/guides/provider_guide.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+## Delphix Provider Guide
+[We have provided a handful of examples in our GitHub repository to help you get a jump start with our Delphix Provider.](https://github.com/delphix-integrations/terraform-provider-delphix/tree/main/examples) These examples range from Terraform resource templates, quick Terraform examples, and full Jenkins pipelines. We update this repository based on our implementation experience, so be sure to check back for updates!
+If you have a suggested template, request, or modification, please submit it in our [GitHub Issues section.](https://github.com/delphix-integrations/terraform-provider-delphix)
```
Comment (nick-mathison) [inline docs/guides/provider_guide.md:5]: Add a new line between 4 and 5

File: examples/jenkins-integration/jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    steps{
+                        echo ('Provisioning Test App...')
+                        sleep (30)
+                    }
+                }
+                stage ('Create Database w/ Terraform') {
+                    steps {
+                        script {
+                            echo ('Provisioning  VDB...')
+                            git branch: 'main', changelog: false, poll: false, url: 'https://github.com/nick-mathison/delphix-terraform-examples.git'
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:34]: In a follow-up, we should update this to reference this folder. Not my personal one. I'll want to shut that one down.
We can handle that in a follow-up.

File: examples/vdb/hana/bookmark/main.tf [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  })
+  appdata_config_params jsonencode({
+    param = "value"
+  })
+  additional_mount_points = [{
+      shared_path = "/",
+      mount_path = "/work",
+      environment_id = "environment-123"
+  }]
+  make_current_account_owner = true
```
Comment (nick-mathison) [inline examples/vdb/hana/bookmark/main.tf:58]: Move tags and `make_current_account_owner` to the bottom. This applies to every example.

The hooks are related to the VDB itself, so they should be closer to the VDB config.

The tags and `make_current..` are more related to DCT configuration and permissions.

File: examples/vdb/hana/bookmark/main.tf [hunk 5]
```diff
@@ -0,0 +1,140 @@
+terraform {
```
Comment (nick-mathison) [inline examples/vdb/hana/bookmark/main.tf:1]: We should add a simple description to the top of these examples files. I think we wrote some during our original implementation?

Suggestion:
```
/**
* Summary: This template showcases the properties available when provisioning an SAP Hana database from a DCT bookmark.
*/
```

**Review by nick-mathison — CHANGES_REQUESTED** [review]
Please add this folder too: https://github.com/nick-mathison/delphix-terraform-examples/tree/main/simple-provision

File: examples/jenkins-integration/jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+                            // sh ('ls -R')
+                            sh ('terraform -chdir=simple-provision init')
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:36]: Update chdir value to if you bring in the full folder

`examples/simple-provision`

File: examples/jenkins-integration/jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                            withCredentials([string(credentialsId: 'DCT_API_KEY', variable: 'KEY')]) {
+                                sh ('terraform -chdir=simple-provision apply -var="source_data_id_1=$SOURCE_VDB" -var="dct_hostname=$DCT_HOSTNAME" -var="dct_api_key=$KEY" -auto-approve')
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:38]: Update chdir value to if you bring in the full folder

`examples/simple-provision`

File: examples/jenkins-integration/jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+                            }
+                            vdb_id = sh(script: 'terraform -chdir=simple-provision output vdb_id_1', returnStdout: true)
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:40]: Update chdir value to if you bring in the full folder

`examples/simple-provision`

File: examples/jenkins-integration/jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+                            vdb_id = vdb_id.replaceAll('\\"', "").trim()
+                            vdb_name = sh(script: 'terraform -chdir=simple-provision output vdb_name_1', returnStdout: true)
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:42]: Update chdir value to if you bring in the full folder

`examples/simple-provision`

File: examples/jenkins-integration/jenkinsfile [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+                    }
+                }
+                stage ('Destroy Database w/ Terraform') {
+                    steps {
+                        script {
+                            if (provision_successful) {
+                                    sleep (60)
+                                    echo ('Destroying Test App and VDB...')
+                                    withCredentials([string(credentialsId: 'DCT_API_KEY', variable: 'KEY')]) {
+                                        sh ('terraform -chdir=simple-provision destroy -var="source_data_id_1=$SOURCE_VDB" -var="dct_hostname=$DCT_HOSTNAME" -var="dct_api_key=$KEY" -auto-approve')
```
Comment (nick-mathison) [inline examples/jenkins-integration/jenkinsfile:164]: Update chdir value to if you bring in the full folder

`examples/simple-provision`

**Review by nick-mathison — CHANGES_REQUESTED** [review]

File: examples/vdb/hana/bookmark/main.tf (hunk 5 above)
Comment (Uddipaan-Hazarika) [reply examples/vdb/hana/bookmark/main.tf:1]: yes will do.

**Review by elima-sugunan — CHANGES_REQUESTED** [review]
@Uddipaan-Hazarika Are the passwords used in multiple files, a commonly used one for Delphix. If so, we need to replace that with a sample one.

File: examples/vdb/hana/bookmark/main.tf (hunk 4 above)
Comment (Uddipaan-Hazarika) [reply examples/vdb/hana/bookmark/main.tf:58]: done

File: examples/jenkins-integration/jenkinsfile (hunk 6 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: done

File: examples/jenkins-integration/jenkinsfile (hunk 7 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: done

File: examples/jenkins-integration/jenkinsfile (hunk 8 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: done

File: examples/jenkins-integration/jenkinsfile (hunk 9 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: done

File: examples/jenkins-integration/jenkinsfile (hunk 10 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: done

File: examples/jenkins-integration/jenkinsfile (hunk 3 above)
Comment (Uddipaan-Hazarika) [reply examples/jenkins-integration/jenkinsfile]: I moved the simple-provision contents to the terraform repo and changed the reference here

**Review by nick-mathison — APPROVED** [review]

File: examples/vdb/hana/snapshot/main.tf [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    mountLocation    = "/mnt/bkmrk"
+    configServices = [{
+        "sTenantServicePort" =  "indexserver:30049"
+      },
+      {
+        "sTenantServicePort" =  "xsengine:30052"
+      }]
+    tDatabaseName   = "tfbkmrk"
+    tSystemUserName = "SYSTEM"
+    tSystemUserPassword ="Delphix_123"
```
Comment (elima-sugunan) [inline examples/vdb/hana/snapshot/main.tf:50]: @Uddipaan-Hazarika, Is this a commonly used password?

Comment (Uddipaan-Hazarika) [reply examples/vdb/hana/snapshot/main.tf:50]: This is actually not a password but its a placeholder. I can change this to something like <PASSWORD> and username as <USERNAME>. Does that work?

**Review by elima-sugunan — APPROVED** [review]

**Review by elima-sugunan — APPROVED** [review]

**Review by elima-sugunan — APPROVED** [review]

File: examples/vdb/hana/snapshot/main.tf (hunk 11 above)
Comment (elima-sugunan) [reply examples/vdb/hana/snapshot/main.tf]: Pwd and ip are changed to placeholders

**Review by arunskurian — APPROVED** [review]

**Pull request merged**